sorts/MergeSort.swift

```diff
@@ -0,0 +1,71 @@
+import Foundation
+
+extension Array where Element: Comparable {
+	
+	mutating func mergeSort(by comparison: (Element, Element) -> Bool) {
+		guard self.count > 1 else {
+			return
+		}
+		_mergeSort(from: 0, to: count - 1, by: comparison)
+	}
+	
+	mutating private func _mergeSort(
+		from left: Int,
+		to right: Int,
+		by comparison: (Element, Element) -> Bool
+	) {
+		if left < right {
+			let mid = left + (right - left) / 2
+			_mergeSort(from: 0, to: mid, by: comparison)
+			_mergeSort(from: mid + 1, to: right, by: comparison)
+			_merge(from: left, mid: mid, to: right, by: comparison)
+		}
+	}
+	
+	mutating private func _merge(
+		from left: Int,
+		mid: Int,
+		to right: Int,
+		by comparison: (Element, Element) -> Bool
+	) {
+		var copy = [Element](repeating: self[left], count: right - left + 1)
+		var (p, q, k) = (left, mid + 1, 0)
```

Comment:
What does `p, q, k` stand for?

Reply:
done, changed variable names to more descriptive names and also, added some visual representation in comments.